akka-stream/src/main/scala/akka/stream/impl/ActorMaterializerImpl.scala

```diff
@@ -244,17 +241,31 @@ private[akka] class SubFusingActorMaterializerImpl(
 
   def takeSnapshotsOfChildren(): Future[immutable.Seq[StreamSnapshot]] = {
     // Arbitrary timeout but should always be quick, the failure scenario is that
-    // the child/stream stopped, and we do retry below
-    implicit val timeout: Timeout = 1.second
-    def takeSnapshot() = {
-      val futureSnapshots =
-        context.children.toList.map(child => (child ? ActorGraphInterpreter.Snapshot).mapTo[StreamSnapshot])
-      Future.sequence(futureSnapshots)
-    }
-
+    // the child/stream stopped, and we do retry below.
     // If the timeout hits it is likely because one of the streams stopped between looking at the list
     // of children and asking it for a snapshot. We retry the entire snapshot in that case
-    retry(() => takeSnapshot(), 3)
+    val children = context.children.toList
+    def takeSnapshot(): Future[List[StreamSnapshot]] = {
+      implicit val ec: ExecutionContext = ExecutionContexts.parasitic
+      val futureSnapshots = children.map(child => {
+        val futureSnapshotOfChild = RetrySupport.retry(() => {
```

Comment:
What would a retry now help with? I.e. how could a message get lost? It could only get delayed in which case a higher timeout might be enough (if you care about that).